org-roam-db.el

```diff
@@ -262,6 +262,31 @@ If the file does not have any connections, nil is returned."
          (files (mapcar 'car-safe (emacsql (org-roam-db) query file))))
     files))
 
+(defun org-roam-db--links-with-max-distance (file max-distance)
+  "Return all files reachable from/connected to FILE in at most MAX-DISTANCE steps,
+including the file itself.  If the file does not have any connections, nil is returned."
+  (let* ((query "WITH RECURSIVE
+                   links_of(file, link) AS
+                     (SELECT \"from\", \"to\" FROM links UNION
+                      SELECT \"to\", \"from\" FROM links),
+                   -- Links are traversed in a breadth-first search.  In order to calculate the
+                   -- distance of nodes and to avoid following cyclic links, the visited nodes
+                   -- are tracked in 'trace'.
```

Comment:
This "trace" is not the most beautiful solution, but what I came up with given little time and the constraints of SQLite. :man_shrugging: I think it should not be too bad, as this function should in any case only be called when the distance (and thus the number of entries in this array) has an upper bound.